core/tests/protractor_utils/ExplorationEditorPage.js

```diff
@@ -97,12 +97,92 @@ var ExplorationEditorPage = function() {
   var saveDiscardToggleButton = element(
     by.css('.protractor-test-save-discard-toggle'));
   var saveDraftButton = element(by.css('.protractor-test-save-draft-button'));
+  var publishExplorationButton = element(
+    by.css('.protractor-test-publish-exploration'));
 
   /*
    * Workflows
    */
   // CONTROLS
 
+  this.publishCardExploration = function(
+      title, objective, category, language, tags) {
+    waitFor.elementToBeClickable(publishExplorationButton,
+      'Publish button taking too long to be clickable.');
+    publishExplorationButton.click();
+
+    var expTitle = element(by.css(
+      '.protractor-test-exploration-title-input'));
+    var expObjective = element(by.css(
+      '.protractor-test-exploration-objective-input'));
+
+    waitFor.elementToBeClickable(expTitle,
+      'Exploration Title input is taking too long to appear');
+
+    expTitle.sendKeys(title);
+    expObjective.sendKeys(objective);
+
+    element(by.css('.select2-container')).click();
+    element(by.css('.select2-dropdown')).element(
+      by.css('.select2-search input')).sendKeys(category + '\n');
+    element(by.css('.protractor-test-exploration-language-select')).click();
+    element(by.css('.protractor-test-exploration-language-select'))
+      .sendKeys(language + '\n');
+    const expTags = element(by.css('.protractor-test-tags'));
+    expTags.click();
+    const expInput = expTags.element(by.tagName('input'));
+    expInput.click();
+    tags.forEach(function(element) {
+      expInput.sendKeys(element, protractor.Key.ENTER);
+    });
+
+    const saveChangesButton = element(by.css(
```

Comment:
This block is not very readable. Can you add newlines in between logical block of statements?

Reply:
Done!